debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/dialect/GeneralDatabaseDialect.java

```diff
@@ -712,13 +713,13 @@ protected String columnQueryBindingFromField(String fieldName, TableDescriptor t
         final ColumnDescriptor column = table.getColumnByName(columnName);
 
         final Object value;
-        if (record.getNonKeyFieldNames().contains(fieldName)) {
+        if (record.getPayload().schema().fields().stream().map(Field::name).toList().contains(fieldName)) {
```

Comment:
This is called for every column when we construct an insert statement. I am concerned with the potential GC and overhead that constantly creating an intermediate list to apply a contains check may incur rather than the contains on a precomputed list we used previously.